project_euler/problem_54.py

```diff
@@ -0,0 +1,156 @@
+        return result_map  # No combinations
+
+
+def check_flush(hand):
+    regexp = "^\w%s(?: \w%s){4}$"
+    for color in colors:
+        if re.search(regexp % (color, color), str(hand)):
+            return True
+    return False
+
+
+def check_straight(hand):
+    for iteration in range(9):
+        sub_card = cards[iteration: iteration + 5]
+        if all(card in hand for card in sub_card[0: 4]):
+            return cards.index(sub_card[4])
+    if all(card in hand for card in "2345A"):
+        return cards.index('5')
+
+
+def check_card_series(hand, range):
+    for card in cards:
+        look_up = '%s[CDHS]' % card
+        rx = re.compile(look_up)
+        if len(rx.findall(hand)) == range:
+            return cards.index(card)
+    return False
+
+
+def check_full_house(hand):
+    pair = False
+    triple = False
+    result_set = []
+    for card in cards:
+        look_up = '%s[CDHS]' % card
+        rx = re.compile(look_up)
+        if len(rx.findall(hand)) == 3:
+            triple = card
+            result_set.append(cards.index(triple))
+        elif len(rx.findall(hand)) == 2:
+            pair = card
+            result_set.append(cards.index(pair))
+    if pair and triple:
+        return result_set
+
+
+def check_2_pairs(hand):
+    first_pair = False
+    second_pair = False
+    result_set = []
+    for card in cards:
+        look_up = '%s[CDHS]' % card
+        rx = re.compile(look_up)
+        if len(rx.findall(hand)) == 2 and not first_pair:
+            first_pair = card
+            result_set.append(cards.index(first_pair))
+        elif len(rx.findall(hand)) == 2 and first_pair:
+            second_pair = card
+            result_set.append(cards.index(second_pair))
+    if first_pair and second_pair:
+        return sorted(result_set, reverse=True)
+
+
+def sort_by_highest_card_value(hand):
+    result_set = []
+    look_up = '(.)[CDHS]'
+    rx = re.compile(look_up)
+    for n in rx.findall(hand):
+        result_set.append(cards.index(n))
+    return sorted(result_set, reverse=True)
+
+
+def compare_hands(first_hand, second_hand):
+    first_hand = evaluate(first_hand)
+    second_hand = evaluate(second_hand)
+    if first_hand['combination'] < second_hand['combination']:
+        return 1  # first hand win
+    elif first_hand['combination'] > second_hand['combination']:
+        return -1  # second hand win
+    elif first_hand['combination'] == second_hand['combination']:
+        if first_hand['additional indicator'] > second_hand['additional indicator']:
+            return 1  # first hand win
+        elif first_hand['additional indicator'] < second_hand['additional indicator']:
+            return -1  # second hand win
+        else:
+            return 0  # draw
+
+
+def main():
+    count_first_win = 0
+    file = open('resource/poker.txt', 'r')
+    for line in file:
+        first_hand, second_hand = line[:len(line) / 2], line[len(line) / 2:]
```

Comment:
You probably want to [strip](https://docs.python.org/2/library/string.html#string.rstrip)  whitespace from the strings, otherwise the regex above may not work.